### ISSUE TITLE
As far as I'm concerned, this is implemented completely wrong.

### ISSUE THREAD

**Issue by damoncasale** [ludohenin/gulp-inline-ng2-template#65]
Templates are supposed to be declared relative to the individual folders that the template files are in, whereas this plugin expects them to be declared relative to the root module.  That's wrong, and should be fixed.

And yes, UseRelativePaths is set to false and I'm still seeing that behavior.  I looked at the code and there's no possible way for it to properly detect the paths of templates in subfolders, relative to their components.

In getFilePath, absPath simply cannot determine the correct relative path.  file.path is always the path of app.module.ts, rather than the path of the related component.

**Comment by ludohenin** [comment]
If you thing something is wrong you are very much welcome to send a PR.

**Comment by gregjacobs** [comment]
Yes, this is definitely an issue when using relative html/css paths in components.

For instance, my component looks like this:

```ts
// app/my-component/my.component.ts
@Component( {
    selector: 'my-component',
    moduleId: __moduleName,
    templateUrl: './my.component.html',
    styleUrls: [ './my.component.css' ]
} )
export class MyComponent {}
```

Here, `gulp-inline-ng2-template` looks for `./my.component.html` instead of `./app/my-component/my.component.html`

**Comment by gregjacobs** [comment]
Err, never mind, totally missed the `useRelativePaths` option. Works as expected!

**Comment by damoncasale** [comment]
That's odd.  useRelativePaths didn't work for me, setting it to either true or false.

**Comment by gforceg** [comment]
This solved the problem for me too!
It seems to me that it would be more intuitive if useRelativePaths true by default, but now I know.

**Comment by unsafePtr** [comment]
How to deal with systemjs and module.id property in angular component? Setting UseRelativePaths doesn't help - `base: "/appTS", target: "es5", UseRelativePaths: true,`. The plugin does not look at sub-folders and try to find all templates in the base folder.

**Comment by gregjacobs** [comment]
@unsafePtr Make sure `useRelativePaths` has a lowercase `u`

**Comment by unsafePtr** [comment]
@gregjacobs Thanks. It was [source ](http://stackoverflow.com/a/37874643/6700082)mistake where is written with big _U_.
